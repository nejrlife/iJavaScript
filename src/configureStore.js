import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import app from './reducers';

import dataSaga from './sagas';

const sagaMiddleware = createSagaMiddleware({
  onError: (error, action) => {
    if (error) {
      console.log('ERROR');
      console.log(error);
      console.log(action);
    }
  }
});

export default function configureStore() {
  const store = createStore(app, applyMiddleware(sagaMiddleware));
  sagaMiddleware.run(dataSaga);
  return store;
}