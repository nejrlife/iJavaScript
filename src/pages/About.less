.aboutContainer {
  width: 688px;
  height: auto;
  background-color: #F2F2F2;
  padding:20px;
  margin-left:20px;
}
.aboutHeader {
  font-family: Microsoft JhengHei;
  font-size: 32px;
  margin: 0;
}
.placeholderFlex {
  display: flex;
  flex-direction: row;
}
.placeholderScaled {
  width: 240px; /* you can use % */
  height: auto;
}
.placeholderFlex p {
  margin-left: 20px;
}
.reusableComponentHeader {
  font-family: Microsoft JhengHei;
  font-size: 18px;
}
.reusableComponentFlex {
  display: flex;
  flex-direction: row;
  gap: 20px;
}

@media (min-width: 768px) and (max-width: 1024px) {
  .aboutContainer {
    width: initial;
    height: auto;
    padding: 20px;
    margin-left: 0px;
  }
}

@media (max-width: 767px) {
  .aboutContainer {
    width: initial;
    height: 100%;
    overflow: auto;
    padding: 20px;
    margin-left: 0px;
  }
  .placeholderScaled {
    width: 100px; /* you can use % */
    height: 100px;
  }
}