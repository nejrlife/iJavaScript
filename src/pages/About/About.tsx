import "./About.less";
import placeholder from "../../assets/img/static_placeholder.jpg";
import ReusableCard from "../../components/ReusableCard/ReusableCard";

const About = () => {
  return (
    <>
      <div className='grey-box aboutContainer'>
        <h2 className='aboutHeader'>
          About iJavaScript
        </h2>
        <hr />
        <div className='placeholderFlex'>
          <img className='placeholderScaled' src={placeholder} alt="placeholder" />
          <p>
            iJavaScript is a research group, founded to check different javascript frameworks based on different design patterns. <br /><br />During Kick-off it will test React, Angular, Knockout and Ember JS features. This use case will prove different aspects of framework capabilities.
          </p>
        </div>
        <h3 className='reusableComponentHeader'>
          Re-usable Component
        </h3>
        <div className='reusableComponentFlex'>
          <ReusableCard key="card1"/>
          <ReusableCard key="card2"/>
        </div>
      </div>
    </>
  )
}

export default About;