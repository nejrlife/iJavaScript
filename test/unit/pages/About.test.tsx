import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import About from '../../../src/pages/About/About';
import { MemoryRouter } from 'react-router-dom';

const initMockRetrieveUserDetailsCustomerState = {
  id: 'IJ101',
  name: 'Tom Jha',
  last_login: '04 Feb 2018',
  balance: '£3000',
  transaction: [
    {
      amount: '£700',
      date: '01 Feb 2018',
      desc: 'Room Rent'
    },
    {
      amount: '£40',
      date: '02 Feb 2018',
      desc: 'Morrison'
    }
  ]
};

const initMockAuthenticateUserDetailsState = {
  userToken: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjdXN0b21lcklkIjoiSUoxMDEiLCJpYXQiOjE2OTUyMDQ2ODksImV4cCI6MTY5NTIwNDY5OX0.eQgS_J9LYjLLJHSIEkM9R6rFcCpGJs0fhRtzxYYEhQk',
  customerId: 'IJ101',
  isUserAuthenticated: true,
  logInError: '',
  logInPending: false,
  isAuthenticatedError: '',
  isAuthenticatedPending: false
};

const INITIAL_STATE = {
  authenticateUser: { ...initMockAuthenticateUserDetailsState },
  retrieveUserDetails: {
    customer: { ...initMockRetrieveUserDetailsCustomerState },
    retrieveUserDetailsError: '',
    retrieveUserDetailsPending: false
  }
}

const mockStore = configureStore([]);

test('should render About page', () => {
  const store = mockStore(INITIAL_STATE);
  const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <About />
        </Provider>
      </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});

test('Fix import React issue', () => {
  console.log(React);
});