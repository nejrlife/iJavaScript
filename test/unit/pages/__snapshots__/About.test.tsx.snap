// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render About page 1`] = `
<div>
  <div
    class="grey-box aboutContainer"
  >
    <h2
      class="aboutHeader"
    >
      About iJavaScript
    </h2>
    <hr />
    <div
      class="placeholderFlex"
    >
      <img
        alt="placeholder"
        class="placeholderScaled"
      />
      <p>
        iJavaScript is a research group, founded to check different javascript frameworks based on different design patterns. 
        <br />
        <br />
        During Kick-off it will test React, Angular, Knockout and Ember JS features. This use case will prove different aspects of framework capabilities.
      </p>
    </div>
    <h3
      class="reusableComponentHeader"
    >
      Re-usable Component
    </h3>
    <div
      class="reusableComponentFlex"
    >
      <div
        class="yellow card"
      />
      <div
        class="yellow card"
      />
    </div>
  </div>
</div>
`;
